### ISSUE TITLE
[expo-file-system] On Android, methods are not being executed, stuck forever (downloadAsync)

### ISSUE THREAD

**Issue by gmartpad** [expo/expo#29219]
### Minimal reproducible example

https://github.com/expo/expo

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

Basically, while running my app on Android, when I try to execute a function that either

- calls downloadAsync from a downloadResumable
- calls downloadAsync from FileSystem

Both start executing but never finish, loading forever.

It doesn't freeze my app, I can continually navigate through screens and click on the screen with feedback, but the downloadAsync function itself never resolves. And nothing in the function that contains the downloadAsync execution, that is present after the downloadAsync ever gets executed either.

It gets to the downloadAsync and then nothing after the method runs.

And tried to execute other methods from FileSystem, they all fail aswell.

```
const downloadCertificate = useCallback(async () => {
    setDownloading(true)
    setDownloadProgress(null)
    try {
      const { access_token } = await localStorage.getItem('access-token')
      setCertificate(certificate)
      const downloadResumable = FileSystem.createDownloadResumable(
        `${urlSubDomain}${certificate.link}${access_token}`,
        `${FileSystem.documentDirectory}${certificate.id}.pdf`,
        {},
        (data) => {
          setDownloadProgress(
            data.totalBytesExpectedToWrite > 0
              ? data.totalBytesWritten / data.totalBytesExpectedToWrite
              : 0
          )
        }
      )

      const file = await downloadResumable.downloadAsync() // <= downloadCertificate callback gets stuck here forever
      if (file === undefined) {
        throw new Error('An error occured when generating the certificate PDF')
      }

      const uri = file.uri

      const { exists } = await FileSystem.getInfoAsync(uri)
      if (!exists) {
        await FileSystem.deleteAsync(uri)
        throw new Error('An error occured when generating the certificate PDF')
      }
      setDownloadedFileUris((oldDownloadedFileUris) => {
        const uriArray = oldDownloadedFileUris
        uriArray[certificate.id] = uri
        return uriArray
      })
      showMessage({
        message: 'Certificate downloaded!',
        description: `You can view it now: (${certificate.name})`,
        type: 'success',
        duration: 8000
      })
    } catch (error: any) {
      Alert.alert('Failure in the download', error.message, [{ text: 'OK' }], {
        cancelable: false
      })
      trackEventFull(errorEvents.downloadCertificate, error)
    }
    setDownloading(false)
  }, [certificate, urlSubDomain])
```

The interesting thing is that the same code runs smoothly on iOS

My package.json: 

```
"dependencies": {
    "@amplitude/analytics-react-native": "^1.4.5",
    "@babel/core": "^7.19.3",
    "@emotion/core": "^11.0.0",
    "@emotion/native": "^11.0.0",
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@expo/config-plugins": "^7.8.0",
    "@expo/metro-config": "~0.17.1",
    "@expo/vector-icons": "^14.0.0",
    "@expo/webpack-config": "~19.0.1",
    "@gorhom/bottom-sheet": "^4.5.1",
    "@mdi/react": "^1.6.0",
    "@react-native-async-storage/async-storage": "1.21.0",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "11.1.0",
    "@react-native-picker/picker": "2.6.1",
    "@react-navigation/bottom-tabs": "^6.3.1",
    "@react-navigation/drawer": "^6.4.1",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.9.26",
    "@react-navigation/stack": "^6.2.1",
    "@react-spring/native": "^9.4.5",
    "@types/node": "^18.0.2",
    "@woli/react-fluid": "0.0.9",
    "axios": "^0.27.2",
    "axios-cancel": "^0.2.2",
    "chroma-js": "^2.4.2",
    "color": "^4.2.3",
    "dotenv": "^16.0.1",
    "expo": "~50.0.19",
    "expo-application": "~5.8.4",
    "expo-asset": "~9.0.2",
    "expo-blur": "~12.9.2",
    "expo-camera": "~14.1.3",
    "expo-clipboard": "~5.0.1",
    "expo-constants": "~15.4.6",
    "expo-dev-client": "~3.3.12",
    "expo-device": "~5.9.4",
    "expo-file-system": "~16.0.9",
    "expo-font": "~11.10.3",
    "expo-image-picker": "~14.7.1",
    "expo-intent-launcher": "~10.11.0",
    "expo-linear-gradient": "~12.7.2",
    "expo-linking": "~6.2.2",
    "expo-media-library": "~15.9.2",
    "expo-notifications": "~0.27.7",
    "expo-screen-orientation": "~6.4.1",
    "expo-secure-store": "~12.8.1",
    "expo-sharing": "~11.10.0",
    "expo-splash-screen": "~0.26.5",
    "expo-sqlite": "~13.4.0",
    "expo-updates": "~0.24.13",
    "form-data": "^4.0.0",
    "formik": "^2.2.9",
    "global": "^4.4.0",
    "html-entities": "^2.3.3",
    "jimp": "^0.16.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.3",
    "native-base": "3.4.5",
    "prop-types": "^15.8.1",
    "qs": "^6.10.3",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.73.6",
    "react-native-circular-progress": "^1.3.7",
    "react-native-color-matrix-image-filters": "6.0.9",
    "react-native-elements": "^3.4.2",
    "react-native-error-boundary": "^1.1.13",
    "react-native-flash-message": "^0.4.1",
    "react-native-gesture-handler": "~2.14.0",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-loading-spinner-overlay": "^3.0.0",
    "react-native-masked-text": "^1.13.0",
    "react-native-pager-view": "6.2.3",
    "react-native-paper": "^5.7.2",
    "react-native-picker-select": "^8.0.4",
    "react-native-progress": "^5.0.0",
    "react-native-reanimated": "~3.6.2",
    "react-native-reanimated-carousel": "^3.5.1",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "4.8.2",
    "react-native-safe-area-view": "^1.1.1",
    "react-native-screens": "~3.29.0",
    "react-native-svg": "14.1.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-native-tab-view": "3.1.1",
    "react-native-text-ticker": "^1.14.0",
    "react-native-vector-icons": "^10.0.3",
    "react-native-web": "~0.19.6",
    "react-native-webview": "13.6.4",
    "react-redux": "^8.0.2",
    "react-transition-group": "^4.4.2",
    "redux": "^4.2.0",
    "redux-persist": "^6.0.0",
    "redux-thunk": "^2.4.1",
    "styled-components": "^5.3.5",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/preset-typescript": "^7.17.12",
    "@react-native-community/cli-debugger-ui": "^8.0.0",
    "@types/chroma-js": "^2.1.4",
    "@types/qs": "^6.9.7",
    "@types/styled-components-react-native": "^5.2.0",
    "babel-plugin-inline-dotenv": "^1.7.0",
    "babel-plugin-module-resolver": "^5.0.0",
    "babel-plugin-tsconfig-paths-module-resolver": "^1.0.3",
    "babel-preset-expo": "^10.0.0",
    "cross-env": "^7.0.3",
    "eslint": "^8.16.0",
    "eslint-config-universe": "^11.0.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react-hooks": "^4.5.0",
    "husky": "^8.0.1",
    "lint-staged": "^13.0.3",
    "metro-react-native-babel-transformer": "^0.77.0",
    "prettier": "^2.6.2",
    "react-devtools": "^4.24.7",
    "react-native-bundle-visualizer": "^3.1.1",
    "redux-devtools-extension": "^2.13.8",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.3"
  },
  "resolutions": {
    "electron": "^11.0.1",
    "@types/react": "^18.0.15"
  },
```

Can't show other parts of the code, since it is company code and all.

But feel free to leave questions, depending on the type of the question, maybe I can get clearance to show the necessary info.

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.17.0 - ~/.nvm/versions/node/v18.17.0/bin/node
      Yarn: 1.22.22 - /opt/homebrew/bin/yarn
      npm: 9.6.7 - ~/.nvm/versions/node/v18.17.0/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.2 AI-232.10227.8.2321.11479570
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      @expo/metro-config: ~0.17.1 => 0.17.7 
      @expo/webpack-config: ~19.0.1 => 19.0.1 
      babel-preset-expo: ^10.0.0 => 10.0.2 
      expo: ~50.0.19 => 50.0.19 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.6 => 0.73.6 
      react-native-web: ~0.19.6 => 0.19.9 
    Expo Workflow: managed

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check that packages match versions required by installed Expo SDK
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

**Comment by gmartpad** [comment]
I still need support on this issue, may someone in the dev team help me with this? Thanks!

**Comment by gmartpad** [comment]
One thing I noticed, during additional tests, is that this behavior doesn't happen with SDK 49, but happens with SDK 50.

here is the package.json with SDK 49 (on which the file system methods work): 

```
"dependencies": {
    "@amplitude/analytics-react-native": "^1.4.5",
    "@babel/core": "^7.19.3",
    "@emotion/core": "^11.0.0",
    "@emotion/native": "^11.0.0",
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@expo/config-plugins": "~7.2.2",
    "@expo/metro-config": "~0.10.0",
    "@expo/vector-icons": "^13.0.0",
    "@expo/webpack-config": "~19.0.1",
    "@gorhom/bottom-sheet": "^4.5.1",
    "@mdi/react": "^1.6.0",
    "@react-native-async-storage/async-storage": "1.18.2",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "9.3.10",
    "@react-native-picker/picker": "2.4.10",
    "@react-navigation/bottom-tabs": "^6.3.1",
    "@react-navigation/drawer": "^6.4.1",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.9.26",
    "@react-navigation/stack": "^6.2.1",
    "@react-spring/native": "^9.4.5",
    "@types/node": "^18.0.2",
    "@woli/react-fluid": "0.0.9",
    "axios": "^0.27.2",
    "axios-cancel": "^0.2.2",
    "chroma-js": "^2.4.2",
    "color": "^4.2.3",
    "dotenv": "^16.0.1",
    "expo": "~49.0.21",
    "expo-application": "~5.3.0",
    "expo-asset": "~8.10.1",
    "expo-blur": "~12.4.1",
    "expo-camera": "~13.4.4",
    "expo-clipboard": "~4.3.1",
    "expo-constants": "~14.4.2",
    "expo-dev-client": "~2.4.13",
    "expo-device": "~5.4.0",
    "expo-file-system": "~15.4.5",
    "expo-font": "~11.4.0",
    "expo-image-picker": "~14.3.2",
    "expo-intent-launcher": "~10.7.0",
    "expo-linear-gradient": "~12.3.0",
    "expo-linking": "~5.0.2",
    "expo-media-library": "~15.4.1",
    "expo-notifications": "~0.20.1",
    "expo-screen-orientation": "~6.0.6",
    "expo-secure-store": "~12.3.1",
    "expo-sharing": "~11.5.0",
    "expo-splash-screen": "~0.20.5",
    "expo-sqlite": "~11.3.3",
    "expo-updates": "~0.18.19",
    "form-data": "^4.0.0",
    "formik": "^2.2.9",
    "global": "^4.4.0",
    "html-entities": "^2.3.3",
    "jimp": "^0.16.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.3",
    "native-base": "3.4.5",
    "prop-types": "^15.8.1",
    "qs": "^6.10.3",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.72.10",
    "react-native-circular-progress": "^1.3.7",
    "react-native-color-matrix-image-filters": "6.0.9",
    "react-native-elements": "^3.4.2",
    "react-native-error-boundary": "^1.1.13",
    "react-native-flash-message": "^0.4.1",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-loading-spinner-overlay": "^3.0.0",
    "react-native-masked-text": "^1.13.0",
    "react-native-pager-view": "6.2.0",
    "react-native-paper": "^5.7.2",
    "react-native-picker-select": "^8.0.4",
    "react-native-progress": "^5.0.0",
    "react-native-reanimated": "~3.3.0",
    "react-native-reanimated-carousel": "^3.5.1",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "4.6.3",
    "react-native-safe-area-view": "^1.1.1",
    "react-native-screens": "~3.22.0",
    "react-native-svg": "13.9.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-native-tab-view": "3.1.1",
    "react-native-text-ticker": "^1.14.0",
    "react-native-vector-icons": "^10.0.3",
    "react-native-web": "~0.19.6",
    "react-native-webview": "13.2.2",
    "react-redux": "^8.0.2",
    "react-transition-group": "^4.4.2",
    "redux": "^4.2.0",
    "redux-persist": "^6.0.0",
    "redux-thunk": "^2.4.1",
    "styled-components": "^5.3.5",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/preset-typescript": "^7.17.12",
    "@react-native-community/cli-debugger-ui": "^8.0.0",
    "@types/chroma-js": "^2.1.4",
    "@types/qs": "^6.9.7",
    "@types/styled-components-react-native": "^5.2.0",
    "babel-plugin-inline-dotenv": "^1.7.0",
    "babel-plugin-module-resolver": "^5.0.0",
    "babel-plugin-tsconfig-paths-module-resolver": "^1.0.3",
    "babel-preset-expo": "^9.5.0",
    "cross-env": "^7.0.3",
    "eslint": "^8.16.0",
    "eslint-config-universe": "^11.0.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react-hooks": "^4.5.0",
    "husky": "^8.0.1",
    "lint-staged": "^13.0.3",
    "metro-react-native-babel-transformer": "^0.77.0",
    "prettier": "^2.6.2",
    "react-devtools": "^4.24.7",
    "react-native-bundle-visualizer": "^3.1.1",
    "redux-devtools-extension": "^2.13.8",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.3"
  },
  "resolutions": {
    "electron": "^11.0.1",
    "@types/react": "^18.0.15"
  },
```

here is the package.json with the SDK 50 (where the file system methods don't work):

```
"dependencies": {
    "@amplitude/analytics-react-native": "^1.4.5",
    "@babel/core": "^7.19.3",
    "@emotion/core": "^11.0.0",
    "@emotion/native": "^11.0.0",
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@expo/config-plugins": "^7.8.0",
    "@expo/metro-config": "~0.17.1",
    "@expo/vector-icons": "^14.0.0",
    "@expo/webpack-config": "~19.0.1",
    "@gorhom/bottom-sheet": "^4.5.1",
    "@mdi/react": "^1.6.0",
    "@react-native-async-storage/async-storage": "1.21.0",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "11.1.0",
    "@react-native-picker/picker": "2.6.1",
    "@react-navigation/bottom-tabs": "^6.3.1",
    "@react-navigation/drawer": "^6.4.1",
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.9.26",
    "@react-navigation/stack": "^6.2.1",
    "@react-spring/native": "^9.4.5",
    "@types/node": "^18.0.2",
    "@woli/react-fluid": "0.0.9",
    "axios": "^0.27.2",
    "axios-cancel": "^0.2.2",
    "chroma-js": "^2.4.2",
    "color": "^4.2.3",
    "dotenv": "^16.0.1",
    "expo": "~50.0.18",
    "expo-application": "~5.8.4",
    "expo-asset": "~9.0.2",
    "expo-blur": "~12.9.2",
    "expo-camera": "~14.1.3",
    "expo-clipboard": "~5.0.1",
    "expo-constants": "~15.4.6",
    "expo-dev-client": "~3.3.12",
    "expo-device": "~5.9.4",
    "expo-file-system": "~16.0.9",
    "expo-font": "~11.10.3",
    "expo-image-picker": "~14.7.1",
    "expo-intent-launcher": "~10.11.0",
    "expo-linear-gradient": "~12.7.2",
    "expo-linking": "~6.2.2",
    "expo-media-library": "~15.9.2",
    "expo-notifications": "~0.27.8",
    "expo-screen-orientation": "~6.4.1",
    "expo-secure-store": "~12.8.1",
    "expo-sharing": "~11.10.0",
    "expo-splash-screen": "~0.26.5",
    "expo-sqlite": "~13.4.0",
    "expo-updates": "~0.24.13",
    "form-data": "^4.0.0",
    "formik": "^2.2.9",
    "global": "^4.4.0",
    "html-entities": "^2.3.3",
    "jimp": "^0.16.1",
    "lodash": "^4.17.21",
    "moment": "^2.29.3",
    "native-base": "3.4.5",
    "prop-types": "^15.8.1",
    "qs": "^6.10.3",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.73.6",
    "react-native-circular-progress": "^1.3.7",
    "react-native-color-matrix-image-filters": "6.0.9",
    "react-native-elements": "^3.4.2",
    "react-native-error-boundary": "^1.1.13",
    "react-native-flash-message": "^0.4.1",
    "react-native-gesture-handler": "~2.14.0",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-keyboard-aware-scroll-view": "^0.9.5",
    "react-native-loading-spinner-overlay": "^3.0.0",
    "react-native-masked-text": "^1.13.0",
    "react-native-pager-view": "6.2.3",
    "react-native-paper": "^5.7.2",
    "react-native-picker-select": "^8.0.4",
    "react-native-progress": "^5.0.0",
    "react-native-reanimated": "~3.6.2",
    "react-native-reanimated-carousel": "^3.5.1",
    "react-native-render-html": "^6.3.4",
    "react-native-safe-area-context": "4.8.2",
    "react-native-safe-area-view": "^1.1.1",
    "react-native-screens": "~3.29.0",
    "react-native-svg": "14.1.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-native-tab-view": "3.1.1",
    "react-native-text-ticker": "^1.14.0",
    "react-native-vector-icons": "^10.0.3",
    "react-native-web": "~0.19.6",
    "react-native-webview": "13.6.4",
    "react-redux": "^8.0.2",
    "react-transition-group": "^4.4.2",
    "redux": "^4.2.0",
    "redux-persist": "^6.0.0",
    "redux-thunk": "^2.4.1",
    "styled-components": "^5.3.5",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/preset-typescript": "^7.17.12",
    "@react-native-community/cli-debugger-ui": "^8.0.0",
    "@types/chroma-js": "^2.1.4",
    "@types/qs": "^6.9.7",
    "@types/styled-components-react-native": "^5.2.0",
    "babel-plugin-inline-dotenv": "^1.7.0",
    "babel-plugin-module-resolver": "^5.0.0",
    "babel-plugin-tsconfig-paths-module-resolver": "^1.0.3",
    "babel-preset-expo": "^10.0.0",
    "cross-env": "^7.0.3",
    "eslint": "^8.16.0",
    "eslint-config-universe": "^11.0.0",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-react-hooks": "^4.5.0",
    "husky": "^8.0.1",
    "lint-staged": "^13.0.3",
    "metro-react-native-babel-transformer": "^0.77.0",
    "prettier": "^2.6.2",
    "react-devtools": "^4.24.7",
    "react-native-bundle-visualizer": "^3.1.1",
    "redux-devtools-extension": "^2.13.8",
    "ts-node": "^10.9.1",
    "typescript": "^5.1.3"
  },
  "resolutions": {
    "electron": "^11.0.1",
    "@types/react": "^18.0.15"
  },
```

**Comment by gmartpad** [comment]
anyone with the same error?

**Comment by ilfoxo** [comment]
Im having the same issue with SDK 51. After some operations with FIleSystem it gets stuck.

In my case I thought that the issue was downloadAsync, but before executing the download i tried to access the FileSystem listing files from my FileSystem.documentDirectory and it got stuck too.

same like you, the app keeps working but any operation over the FileSystem get stuck.

For additional information, the first time my app runs execute a configuration process:

1) Download some image files and save it to FileSystem.documentDirectory.
2) Create a sqlite database (in FileSystem.documentDirectory) and some tables.
3) Get information from an API a save it to database.
4) Download image items from cloud to FileSystem.documentDirectory

FileSystem get stuck sometimes in points 2, 3 or 4.

Maybe it's stupid what Im going to say, but it looks like if I access FileSystem many times it gets stuck and the only way to recover is to restart the app and fails again.

Next, my package.json:

{
  "name": "",
  "version": "1.0.0",
  "main": "expo-router/entry",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@aws-amplify/react-native": "^1.1.1",
    "@aws-amplify/ui-react-native": "^2.2.2",
    "@gorhom/bottom-sheet": "4",
    "@hookform/resolvers": "^3.4.2",
    "@react-native-async-storage/async-storage": "^1.23.1",
    "@react-native-community/datetimepicker": "8.0.1",
    "@react-navigation/drawer": "^6.6.15",
    "@shopify/flash-list": "1.6.4",
    "@tabler/icons": "^3.10.0",
    "@tabler/icons-react-native": "^3.5.0",
    "@tanstack/react-query": "^5.40.0",
    "aws-amplify": "^6.3.4",
    "crypto-es": "^2.1.0",
    "expo": "~51.0.17",
    "expo-clipboard": "~6.0.3",
    "expo-constants": "~16.0.2",
    "expo-document-picker": "~12.0.2",
    "expo-file-system": "~17.0.1",
    "expo-linking": "~6.3.1",
    "expo-print": "~13.0.1",
    "expo-router": "~3.5.17",
    "expo-secure-store": "~13.0.2",
    "expo-sharing": "~12.0.1",
    "expo-sqlite": "~14.0.4",
    "expo-status-bar": "~1.12.1",
    "install": "^0.13.0",
    "lodash.debounce": "^4.0.8",
    "nativewind": "^2.0.11",
    "npm": "^10.8.1",
    "react": "18.2.0",
    "react-content-loader": "^7.0.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.51.5",
    "react-native": "0.74.3",
    "react-native-csv": "^0.2.0",
    "react-native-gesture-handler": "~2.16.1",
    "react-native-get-random-values": "^1.11.0",
    "react-native-reanimated": "~3.10.1",
    "react-native-safe-area-context": "4.10.1",
    "react-native-screens": "3.31.1",
    "react-native-svg": "15.2.0",
    "react-native-webview": "13.8.6",
    "undefined": "@hookform/resolvers/zod",
    "zod": "^3.23.8",
    "zustand": "^4.5.2",
    "@react-native-community/netinfo": "11.3.2"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@types/lodash.debounce": "^4.0.9",
    "@types/react": "~18.2.45",
    "tailwindcss": "^3.3.2",
    "typescript": "~5.3.3"
  },
  "private": true
}

**Comment by gmartpad** [comment]
@ilfoxo any update on your situation?

**Comment by ilfoxo** [comment]
I solved my issue, but i dont really know what was the real action. The changes that I made:

1) I was working with expo-sqlite-legacy. I update all the code to work with expo-sqlite
2) I organized the connections to sqlite database open/close connections only when necessary.

What I suspect is that to leave open connections to database blocks the filesystem, **it is only an assumption**.

I hope this information helps to solve your issue. Check your code, i had to rewrite lots of it...
